substrate/primitives/runtime/src/lib.rs

```diff
@@ -479,6 +501,13 @@ impl Verify for MultiSignature {
 				sp_io::crypto::secp256k1_ecdsa_recover_compressed(sig.as_ref(), &m)
 					.map_or(false, |pubkey| sp_io::hashing::blake2_256(&pubkey) == who)
 			},
+			Self::Eth(sig) => {
+				let m = sp_io::hashing::blake2_256(msg.get());
```

Comment:
I kinda feel this is a "now or never" scenario, so here's my two cents:

Why not hashing with keccak? Reasons:

1) Adhere to the Ethereum standard completely.
2) Enable pure Ethereum wallets to sign transactions on Polkadot. Yes, initially that could be blind signing, but it opens the door to further improvements on the field.

Reply:
I also thought about this. I decided against it because I thought if they support our transactions they also support blake. And I tried to make it most similar to the other ecdsa signature type.

However, you have a good point with blind signing. We could even sign our transactions using EIP-712 without having explicit wallet support.

So yeah I think we should use keccak here.

Reply:
Okay for this to work we also need to make changes to the ecdsa::Pair which is the same for both signers. Cause that defines how we sign the key. I will need to look into this.

Reply:
Yeah something like that. The problem is to find a way to accomplish this change without causing a huge diff.

Reply:
Thanks to you! 🙏